contrib/executor/jmeterd/pkg/runner/runner.go

```diff
@@ -0,0 +1,347 @@
+	}
+
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return result, err
+	}
+
+	if fileInfo.IsDir() {
+		scriptName := execution.Args[len(execution.Args)-1]
+		if workingDir != "" {
+			path = filepath.Join(r.Params.DataDir, "repo")
+			if execution.Content != nil && execution.Content.Repository != nil {
+				scriptName = filepath.Join(execution.Content.Repository.Path, scriptName)
+			}
+		}
+
+		output.PrintLog(fmt.Sprintf("execution arg before %s", execution.Args))
+		execution.Args = execution.Args[:len(execution.Args)-1]
+		output.PrintLog(fmt.Sprintf("execution arg afrer %s", execution.Args))
+		output.PrintLogf("%s It is a directory test - trying to find file from the last executor argument %s in directory %s", ui.IconWorld, scriptName, path)
+
+		// sanity checking for test script
+		scriptFile := filepath.Join(path, workingDir, scriptName)
+		fileInfo, errFile := os.Stat(scriptFile)
+		if errors.Is(errFile, os.ErrNotExist) || fileInfo.IsDir() {
+			output.PrintLogf("%s Could not find file %s in the directory, error: %s", ui.IconCross, scriptName, errFile)
+			return *result.Err(errors.Errorf("could not find file %s in the directory: %v", scriptName, errFile)), nil
+		}
+		path = scriptFile
+	}
+
+	slavesEnvVariables := jmeter_env.ExtractSlaveEnvVariables(envManager.Variables)
+	// compose parameters passed to JMeter with -J
+	params := make([]string, 0, len(envManager.Variables))
+	for _, value := range envManager.Variables {
+		if value.Name == jmeter_env.MasterOverrideJvmArgs || value.Name == jmeter_env.MasterAdditionalJvmArgs {
+			//Skip JVM ARGS to be appended in the command
+			continue
+		}
+		params = append(params, fmt.Sprintf("-J%s=%s", value.Name, value.Value))
+
+	}
+
+	runPath := r.Params.DataDir
+	if workingDir != "" {
+		runPath = workingDir
+	}
+
+	parentTestFolder := filepath.Join(filepath.Dir(path))
+	// Set env plugin env variable to set custom plugin directory
+	// with this path custom plugin will be copied to jmeter's plugin directory
+	err = os.Setenv("JMETER_PARENT_TEST_FOLDER", parentTestFolder)
+	if err != nil {
+		output.PrintLogf("%s Failed to set parent test folder directory %s", ui.IconWarning, parentTestFolder)
+	}
+	// Add user plugins folder in slaves env variables
+	slavesEnvVariables["JMETER_PARENT_TEST_FOLDER"] = testkube.NewBasicVariable("JMETER_PARENT_TEST_FOLDER", parentTestFolder)
+
+	outputDir := filepath.Join(runPath, "output")
+	// clean output directory it already exists, only useful for local development
+	_, err = os.Stat(outputDir)
+	if err == nil {
+		if err = os.RemoveAll(outputDir); err != nil {
+			output.PrintLogf("%s Failed to clean output directory %s", ui.IconWarning, outputDir)
+		}
+	}
+	// recreate output directory with wide permissions so JMeter can create report files
+	if err = os.Mkdir(outputDir, 0777); err != nil {
+		return *result.Err(errors.Errorf("could not create directory %s: %v", runPath, err)), nil
```

Comment:
While doing this change I am getting error because `errors.Wrapf` takes first argument as `err` and second argument as `string` i.e. provided error message. As per my opinion this is not user friendly, so I have just replaced the error message in the `errors.Errorf` as provided by you. like from `could not create directory` to `error creating directory`.